adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/impl/ConfigBasedCoapEndpointFactory.java

```diff
@@ -181,9 +181,12 @@ private NetworkConfig newDefaultNetworkConfig() {
                 } else {
                     LOG.warn("error reading NetworkConfig file [{}]", fileName, readAttempt.cause());
                 }
+                result.complete(networkConfig);
```

Comment:
FMPOV we should either complete or fail the result depending on the outcome of trying to read the file.

Reply:
I considered to implement the behavior close to the behavior before this fix.
But failing will also be a possibility, if the resulting behavior change is acceptable.

Reply:
Well, if I would explicitly configure a config file to be read and used then I would expect the adapter to fail to start, if the file could not be read, because otherwise I would get the impression that it is now using the configuration that I put into the file when it actually doesn't. So FMPOV we should fail the start up if the file cannot be read. Does that make sense to you as well?

Reply:
Sure, that makes sense. My concern was more, if such a behavior change would be acceptable in one commit. It seems so. So I will adapt the PR.

Reply:
done :-).

Reply:
can you also adapt the CoAP adapter admin guide to mention that start up will fail if the configured config file cannot be read/found?

Reply:
Yes. Working on migration to Californium 3.0, I also found, that the warning "skipping malformed NetworkConfig properties" is now also misleading.

Reply:
Done.